src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -29,17 +30,15 @@ import io.grpc.StatusException
  */
 class CosmasGoogleCloudService(private val bucketName: String,
                                private val storage: Storage = StorageOptions.getDefaultInstance().service) : CosmasGrpc.CosmasImplBase() {
-
-    private val fileBuffer = mutableMapOf<String, MutableMap<String, ByteString>>().withDefault { mutableMapOf() }
+    private val fileBuffer = ConcurrentHashMap<String, ConcurrentHashMap<String, ByteString>>().withDefault { ConcurrentHashMap() }
 
     override fun createVersion(request: CosmasProto.CreateVersionRequest,
                                responseObserver: StreamObserver<CosmasProto.CreateVersionResponse>) {
         println("Get request for create new version of file # ${request.fileId}")
-        synchronized(fileBuffer) {
```

Comment:
Hey, I didn't call on removal of synchronized blocks. ConcurrentHashMap is not a replacement of mutex synchronization in general, although it might be a replacement in some cases. It guarantees predictable behavior when two conflicting methods, such as entries() and put() are executed concurrently, but here you have a sequence of calls: 1)get value + 2)do something + 3)put value and ConcurrentHashMap will not help you if some concurrent thread executing the same sequence. The winner will be the one who executes 3) the last which is not what you want.

Reply:
But `commitVersion` doesn't write anything, so I don't have to add synchronized block, right?

Reply:
That's right, in `commitVersion` you don't have to synchronize.

Reply:
ok